tests/crud_operations.js

```diff
@@ -0,0 +1,133 @@
+const Job = require("../models/Job");
+const { factory, seed_db ,testUserPassword } = require("../util/seed_db");
+const get_chai = require("../util/get_chai");
+const { app } = require("../app");
+
+
+before(async () => {
+    const { expect, request } = await get_chai();
+    this.test_user = await seed_db();
+    
+    // Get the login page to extract CSRF token
+    let req = request.execute(app).get("/sessions/logon").send();
+    let res = await req;
+    
+    console.log("Response Text:", res.text);  // Log the response to check CSRF token presence
+    
+    const textNoLineEnd = res.text.replaceAll("\n", "");
+    
+    // Extract CSRF token from the form
+    this.csrfToken = /_csrf"\s+value="(.*?)"/.exec(textNoLineEnd);
+    
+    if (!this.csrfToken) {
+      throw new Error("CSRF Token not found in the response");
```

Comment:
Generally speaking, you aren't going to want to be throwing an error in this way with tests (although that does depend on the framework / library and how you design your tests).